internal/status/proxystatus.go

```diff
@@ -51,7 +49,14 @@ func (pu *ProxyUpdate) ConditionFor(cond ConditionType) *projectcontour.Detailed
 	if !ok {
 		newDc := &projectcontour.DetailedCondition{}
 		newDc.Type = string(cond)
-
+		newDc.ObservedGeneration = pu.Generation
+		if newDc.IsPositivePolarity() {
+			newDc.Status = projectcontour.ConditionTrue
+			newDc.Reason = "Valid"
+			newDc.Message = "Valid HTTPProxy"
```

Comment:
`IsPositivePolarity` treats the condition generically, but then the reason and message assume that the condition was `ValidCondition`. So just be explicit that this expects a `ValidCondition`.